test/e2e/aws_test.go

```diff
@@ -19,112 +19,438 @@ limitations under the License.
+			JoinConfiguration: &kubeadmv1beta1.JoinConfiguration{
+				ControlPlane: &kubeadmv1beta1.JoinControlPlane{},
+				NodeRegistration: kubeadmv1beta1.NodeRegistrationOptions{
+					Name:             "{{ ds.meta_data.hostname }}",
+					KubeletExtraArgs: map[string]string{"cloud-provider": "aws"},
+				},
+			},
+		},
+	}
+	Expect(kindClient.Create(context.TODO(), config)).NotTo(HaveOccurred())
+}
+
+func makeInitBootstrapConfig(namespace, name string) {
+	fmt.Fprintf(GinkgoWriter, "Creating Init KubeadmConfig %s/%s\n", namespace, name)
+	config := &bootstrapv1.KubeadmConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: bootstrapv1.KubeadmConfigSpec{
+			ClusterConfiguration: &kubeadmv1beta1.ClusterConfiguration{
+				APIServer: kubeadmv1beta1.APIServer{
+					ControlPlaneComponent: kubeadmv1beta1.ControlPlaneComponent{
+						ExtraArgs: map[string]string{"cloud-provider": "aws"},
+					},
+				},
+				ControllerManager: kubeadmv1beta1.ControlPlaneComponent{
+					ExtraArgs: map[string]string{"cloud-provider": "aws"},
+				},
+			},
+			InitConfiguration: &kubeadmv1beta1.InitConfiguration{
+				NodeRegistration: kubeadmv1beta1.NodeRegistrationOptions{
+					Name:             "{{ ds.meta_data.hostname }}",
+					KubeletExtraArgs: map[string]string{"cloud-provider": "aws"},
+				},
+			},
+		},
+	}
+	Expect(kindClient.Create(context.TODO(), config)).NotTo(HaveOccurred())
+}
+
+func makeAWSMachine(namespace, name string) {
+	fmt.Fprintf(GinkgoWriter, "Creating AWSMachine %s/%s\n", namespace, name)
+	awsMachine := &infrav1.AWSMachine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: infrav1.AWSMachineSpec{
+			InstanceType:       "t3.large",
+			IAMInstanceProfile: "control-plane.cluster-api-provider-aws.sigs.k8s.io",
+			SSHKeyName:         keyPairName,
+		},
+	}
+	Expect(kindClient.Create(context.TODO(), awsMachine)).NotTo(HaveOccurred())
+}
+
+func makeCluster(namespace, name, awsClusterName string) {
+	fmt.Fprintf(GinkgoWriter, "Creating Cluster %s/%s\n", namespace, name)
+	cluster := &clusterv1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: clusterv1.ClusterSpec{
+			ClusterNetwork: &clusterv1.ClusterNetwork{
+				Pods: &clusterv1.NetworkRanges{
+					CIDRBlocks: []string{"192.168.0.0/16"},
+				},
+			},
+			InfrastructureRef: &corev1.ObjectReference{
+				Kind:       "AWSCluster",
+				APIVersion: infrav1.GroupVersion.String(),
+				Name:       awsClusterName,
+				Namespace:  namespace,
+			},
+		},
+	}
+	Expect(kindClient.Create(context.TODO(), cluster)).NotTo(HaveOccurred())
+}
+
+func makeAWSCluster(namespace, name string) {
+	fmt.Fprintf(GinkgoWriter, "Creating AWSCluster %s/%s\n", namespace, name)
+	awsCluster := &infrav1.AWSCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: infrav1.AWSClusterSpec{
+			Region:     "us-east-1",
```

Comment:
Isn't it better to get the region from env?
I faced this issue while running e2e in my setup.

Reply:
@BhargavMadduru good call, could be something that we handle as a followup to avoid blocking this PR merging for longer if needed, though.

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1161